### ISSUE TITLE
Compilation warnings about OSMemoryBarrier

### ISSUE THREAD

**Issue by elbamos** [RcppCore/RcppParallel#159]
I get a host of warnings about deprecated function calls on compilation: 

```
Library/R/arm64/4.1/library/RcppParallel/include/tbb/machine/macos_common.h:59:13: warning: 'OSAtomicCompareAndSwap64Barrier' is deprecated: first deprecated in macOS 10.12 - Use std::atomic_compare_exchange_strong() from <atomic> instead [-Wdeprecated-declarations]

Library/R/arm64/4.1/library/RcppParallel/include/tbb/machine/macos_common.h:63:28: warning: 'OSAtomicAdd64' is deprecated: first deprecated in macOS 10.12 - Use std::atomic_fetch_add_explicit(std::memory_order_relaxed) from <atomic> instead [-Wdeprecated-declarations]

Library/R/arm64/4.1/library/RcppParallel/include/tbb/tbb_machine.h:612:9: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use std::atomic_thread_fence() from <atomic> instead [-Wdeprecated-declarations]
```

There are others, all referring to calls from macos_common.h or tbb_machine.h, and all referring to deprecation after OS X 10.12.

They don't bother me, except that they show up as a a Warning when running R CMD CHECK to prepare a package for submission to CRAN.  I don't see any mention of them on the Intel TBB pages so I'm reporting it here.

**Comment by kevinushey** [comment]
Are these resolved now with RcppParallel 5.1.2?

**Comment by elbamos** [comment]
Yup, thanks!
